health_checks.go

```diff
@@ -0,0 +1,201 @@
+func (c *Client) HealthCheckCertificateExpiration(within uint, unit TimeUnit) (rec *Health, err error) {
+	req, err := newGETRequest(c, "health/checks/certificate-expiration/"+strconv.Itoa(int(within))+"/"+string(unit))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeAndParseRequest(c, req, &rec); err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+// Responds a 200 OK if there is an active listener on the give port, otherwise responds with a 503 Service Unavailable.
+func (c *Client) HealthCheckPortListenerListener(port uint) (rec *Health, err error) {
+	req, err := newGETRequest(c, "health/checks/port-listener/"+strconv.Itoa(int(port)))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeAndParseRequest(c, req, &rec); err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+// Responds a 200 OK if there is an active listener for the given protocol, otherwise responds with a 503 Service Unavailable.
+// Valid protocol names are: amqp091, amqp10, mqtt, stomp, web-mqtt, web-stomp.
+func (c *Client) HealthCheckProtocolListener(protocol Protocol) (rec *Health, err error) {
+	req, err := newGETRequest(c, "health/checks/protocol-listener/"+string(protocol))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeAndParseRequest(c, req, &rec); err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+// Responds a 200 OK if all virtual hosts and running on the target node, otherwise responds with a 503 Service Unavailable.
+func (c *Client) HealthCheckVirtualHosts() (rec *Health, err error) {
+	req, err := newGETRequest(c, "health/checks/virtual-hosts")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeAndParseRequest(c, req, &rec); err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+// Checks if there are classic mirrored queues without synchronised mirrors online (queues that would potentially lose data if the target node is shut down).
+// Responds a 200 OK if there are no such classic mirrored queues, otherwise responds with a 503 Service Unavailable.
+func (c *Client) HealthCheckNodeIsMirrorSyncCritical() (rec *Health, err error) {
+	req, err := newGETRequest(c, "health/checks/node-is-mirror-sync-critical")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeAndParseRequest(c, req, &rec); err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+// Checks if there are quorum queues with minimum online quorum (queues that would lose their quorum and availability if the target node is shut down).
+// Responds a 200 OK if there are no such quorum queues, otherwise responds with a 503 Service Unavailable.
+func (c *Client) HealthCheckNodeIsQuorumCritical() (rec *Health, err error) {
+	req, err := newGETRequest(c, "health/checks/node-is-quorum-critical")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeAndParseRequest(c, req, &rec); err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+// Deprecated health check api
```

Comment:
I think we should not add any support for the deprecated endpoints.

Reply:
Yes, I will remove it.